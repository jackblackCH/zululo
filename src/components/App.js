import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import ShowCountries from './ShowCountries';
import mainReducer from '../reducers';

const store = createStore(mainReducer);

const App = () => {
	return (
    <Provider store={store}>
		  <ShowCountries />
    </Provider>
	)
}

export default App;
