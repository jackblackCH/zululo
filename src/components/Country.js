import React, { Component } from 'react';

export default class Country extends Component {
    render() {
        return (
            <div>
                <h1>Country</h1>
            </div>
        )
    }
}