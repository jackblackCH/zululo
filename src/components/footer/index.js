import React from 'react';
import './Footer.css'

 export default () => <footer className="c-footer"><a className="c-footer__link" href="//github.com/jackblackCH/zululo"> Copyright  zululo 2017</a></footer>
