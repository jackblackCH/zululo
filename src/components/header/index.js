import React from 'react';
import './Header.css'

export default () => <header className="c-header"><a href="/"><img alt="" className="c-header__logo" src='/assets/logo.png' /></a></header>
