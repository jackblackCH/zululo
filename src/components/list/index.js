import React, {Component} from 'react';
import {connect} from 'react-redux';
import './list.css';

export class List extends Component {
    render() {
        return (
            <ul className="c-list">
                {this.props.countries.map(country => {
                    return (
                        <li className="c-list__item" key={country.shortName}>
                            <div className="c-list__block">
                              <h2 className="c-list__title">{country.name}</h2>
                              <img alt="{country.name}" className="c-list__image" src={country.photo} />
                            </div>
                        </li>
                    )
                })}
            </ul>
        );
    }
}

const mapStateToProps = (state) => {
    const countries = state.countries;
    return {
        countries
    };
};


export default connect(mapStateToProps)(List)
