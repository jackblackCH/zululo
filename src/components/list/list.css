.c-list {
  padding: 0;
  margin: 0; }
  .c-list__item {
    position: relative;
    height: 33.33vh; }
  .c-list__title {
    text-shadow: 4px 4px 10px #969696;
    font-size: 40px;
    text-align: center;
    color: #f3f3f3;
    position: absolute;
    top: 50%;
    width: 100%;
    margin-top: -30px;
    z-index: 2; }
  .c-list__block {
    display: block;
    vertical-align: middle; }
  .c-list__image {
    transition: opacity .25s ease-in-out;
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    opacity: 1;
    height: 100%;
    width: 100%; }
    .c-list__image:hover {
      opacity: 0.8; }
