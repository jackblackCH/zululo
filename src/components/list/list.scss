.c-list {
    padding: 0;
    margin: 0;

    &__item {
      position: relative;
      height: 33.33vh;
    }

    &__title {
      text-shadow: 4px 4px 10px rgba(150, 150, 150, 1);
      font-size: 40px;
      text-align: center;
      color: #f3f3f3;
      position: absolute;
      top: 50%;
      width: 100%;
      margin-top: -30px;
      z-index: 2;
    }

    &__block {
      display: block;
      vertical-align: middle;
    }

    &__image {
      transition: opacity .25s ease-in-out;
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
      opacity: 1;
      height: 100%;
      width: 100%;

      &:hover {
          opacity: 0.8;
      }
    }
}
