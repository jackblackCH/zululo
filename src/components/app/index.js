import React from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import CountryList from '../list/';
import Country from '../Country';
import mainReducer from '../../reducers';
import {
    browserHistory,
    Router,
    Route,
    IndexRoute,
} from "react-router";

const store = createStore(mainReducer);

const App = () => {
    return (
        <Provider store={store}>
            <Router history={browserHistory}>
                <Route path="/" component={({ children }) => <div>{children}</div>}>
                    <Route path="country/:id" component={Country} />
                    <IndexRoute component={CountryList} />
                </Route>
            </Router>
        </Provider>
    )
};

export default App;
