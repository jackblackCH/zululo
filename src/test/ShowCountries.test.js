/**
 * Tests for the TodoItem
 */
 import React from 'react'
 import { List } from '../components/list'
 import renderer from 'react-test-renderer'


 describe('Component: Country list', () => {

   it('should render the list', () => {
     const tree = renderer.create(
       <List countries={[]}></List>
     ).toJSON();
     expect(tree).toMatchSnapshot()
   })

   it('should have a classname', () => {
     const tree = renderer.create(
       <List className='c-list' countries={[]}></List>
     ).toJSON();
     expect(tree).toMatchSnapshot()
   })

   it('should render items', () => {
     const tree = renderer.create(
       <List countries={[{
           name: 'United Kingdom',
           shortName: 'uk',
           photo: '/assets/country/city.jpg'
       },
       {
           name: 'Switzerland',
           shortName: 'ch',
           photo: '/assets/country/switzerlandAlps.jpg'
       },
       {
           name: 'Poland',
           shortName: 'pl',
           photo: '/assets/country/krakow.jpg'
       }]} ></List>
     ).toJSON();
     expect(tree).toMatchSnapshot()
   })
});
