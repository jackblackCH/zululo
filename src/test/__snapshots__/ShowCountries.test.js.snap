exports[`Component: Country list should have a classname 1`] = `
<ul
  className="c-list" />
`;

exports[`Component: Country list should render items 1`] = `
<ul
  className="c-list">
  <li
    className="c-list__item">
    <div
      className="c-list__block">
      <h2
        className="c-list__title">
        United Kingdom
      </h2>
      <img
        className="c-list__image"
        src="/assets/country/city.jpg" />
    </div>
  </li>
  <li
    className="c-list__item">
    <div
      className="c-list__block">
      <h2
        className="c-list__title">
        Switzerland
      </h2>
      <img
        className="c-list__image"
        src="/assets/country/switzerlandAlps.jpg" />
    </div>
  </li>
  <li
    className="c-list__item">
    <div
      className="c-list__block">
      <h2
        className="c-list__title">
        Poland
      </h2>
      <img
        className="c-list__image"
        src="/assets/country/krakow.jpg" />
    </div>
  </li>
</ul>
`;

exports[`Component: Country list should render the list 1`] = `
<ul
  className="c-list" />
`;
